### PR TITLE
Con 3156 confirm edit

### PR DESCRIPTION
**Pull request by hkf-tech**

### CODE REVIEW

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Mappers/Apprentice/ConfirmEditApprenticehsipRequestToConfirmEditViewModelMapper.cs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            {
+                vm.BirthDay =   source.BirthDay;
+                vm.BirthMonth = source.BirthMonth;
+                vm.BirthYear =  source.BirthYear;
+            }
+            vm.OriginalApprenticeship.BirthDay = apprenticeship.DateOfBirth.Day;
+            vm.OriginalApprenticeship.BirthMonth = apprenticeship.DateOfBirth.Month;
+            vm.OriginalApprenticeship.BirthYear = apprenticeship.DateOfBirth.Year;
+
+            if (source.Cost != priceEpisodes.PriceEpisodes.GetPrice())
```
Comment (cofaulco) [inline src/SFA.DAS.EmployerCommitmentsV2.Web/Mappers/Apprentice/ConfirmEditApprenticehsipRequestToConfirmEditViewModelMapper.cs:76]: `priceEpisodes.PriceEpisodes.GetPrice()` we could do this just once?

File: src/SFA.DAS.EmployerCommitmentsV2.Web.UnitTests/Mappers/Apprentice/ConfirmDetailsAndSendViewModelMapperTests - Copy.cs [hunk 2]
```diff
@@ -0,0 +1,250 @@
+﻿using AutoFixture;
```
Comment (cofaulco) [inline src/SFA.DAS.EmployerCommitmentsV2.Web.UnitTests/Mappers/Apprentice/ConfirmDetailsAndSendViewModelMapperTests - Copy.cs:1]: This filename has `- copy` in

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Controllers/ApprenticeController.cs [hunk 3]
```diff
@@ -519,15 +518,41 @@ public async Task<IActionResult> EditApprenticeship(EditApprenticeshipRequestVie
...
+        public async Task<IActionResult> ConfirmEditApprenticeship(ConfirmEditApprenticeshipViewModel viewModel)
+        {
+            if (viewModel.ConfirmChanges.Value)
+            {
+                var request = await _modelMapper.Map<EditApprenticeshipApiRequest>(viewModel);
+                var result = await _commitmentsApiClient.EditApprenticeship(request);
+
+                if (result.NeedReapproval)
+                {
+                    TempData.AddFlashMessage("Suggested changes sent to training provider for approval, where needed.", ITempDataDictionaryExtensions.FlashMessageLevel.Info);
```
Comment (VasanthaKasirajan3008) [inline src/SFA.DAS.EmployerCommitmentsV2.Web/Controllers/ApprenticeController.cs:547]: use the const keyword to declare flash message

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Controllers/ApprenticeController.cs [hunk 4]
```diff
@@ -519,15 +518,41 @@ public async Task<IActionResult> EditApprenticeship(EditApprenticeshipRequestVie
...
+                }
+                else
+                {
+                    TempData.AddFlashMessage("Apprentice updated", ITempDataDictionaryExtensions.FlashMessageLevel.Info);
```
Comment (VasanthaKasirajan3008) [inline src/SFA.DAS.EmployerCommitmentsV2.Web/Controllers/ApprenticeController.cs:551]: same like above

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Mappers/Apprentice/ConfirmEditApprenticehsipViewModelToEditApiRequestMapper.cs [hunk 5]
```diff
@@ -0,0 +1,31 @@
+﻿using SFA.DAS.CommitmentsV2.Api.Types.Requests;
+using SFA.DAS.CommitmentsV2.Shared.Interfaces;
+using SFA.DAS.EmployerCommitmentsV2.Web.Models.Apprentice;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.EmployerCommitmentsV2.Web.Mappers.Apprentice
+{
+    public class ConfirmEditApprenticehsipViewModelToEditApiRequestMapper : IMapper<ConfirmEditApprenticeshipViewModel, EditApprenticeshipApiRequest>
+    {
+        public ConfirmEditApprenticehsipViewModelToEditApiRequestMapper()
```
Comment (VasanthaKasirajan3008) [inline src/SFA.DAS.EmployerCommitmentsV2.Web/Mappers/Apprentice/ConfirmEditApprenticehsipViewModelToEditApiRequestMapper.cs:10]: do we need this ctor?

File: src/SFA.DAS.EmployerCommitmentsV2.Web.UnitTests/Mappers/Apprentice/ConfirmDetailsAndSendViewModelMapperTests - Copy.cs [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+
+            Assert.AreNotEqual(fixture.source.EndDate.Date, fixture._apprenticeshipResponse.EndDate);
+            Assert.AreEqual(fixture.source.EndDate.Date, result.EndDate);
+            Assert.AreEqual(fixture._apprenticeshipResponse.EndDate, result.OriginalApprenticeship.EndDate);
+        }
+
+        [Test]
+        public async Task WhenCourseIsChanged()
+        {
+            fixture.source.CourseCode = "Abc";
```
Comment (VasanthaKasirajan3008) [inline src/SFA.DAS.EmployerCommitmentsV2.Web.UnitTests/Mappers/Apprentice/ConfirmDetailsAndSendViewModelMapperTests - Copy.cs:121]: "Abc" used for coursecode,firstname, lastname, employerref .. can we change  to different related strings please

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Views/Apprentice/ConfirmEditApprenticeship.cshtml [hunk 7]
```diff
@@ -1,7 +1,201 @@
...
+                        <div class="govuk-summary-list__row">
+                            <dt class="govuk-summary-list__key"> Unique learner number </dt>
+                            <dd class="govuk-summary-list__value"> @Model.OriginalApprenticeship.ULN</dd>
+                        </div>
+                        <div class="govuk-summary-list__row">
+                            <dt class="govuk-summary-list__key"> Apprenticeship training course </dt>
+                            <dd class="govuk-summary-list__value"> @Model.OriginalApprenticeship.TrainingName</dd>
+                        </div>
+                    </dl>
+                    @*<table class="govuk-table">
```
Comment (VasanthaKasirajan3008) [inline src/SFA.DAS.EmployerCommitmentsV2.Web/Views/Apprentice/ConfirmEditApprenticeship.cshtml:55]: commented lines can be removed

**Review by cofaulco — APPROVED** [review]

**Review by cofaulco — APPROVED** [review]

**Pull request merged**